src/Build/BackEnd/Components/Logging/LoggingServiceLogMethods.cs

```diff
@@ -738,13 +739,7 @@ public void LogTaskStarted(BuildEventContext taskBuildEventContext, string taskN
         public BuildEventContext LogTaskStarted2(BuildEventContext targetBuildEventContext, string taskName, string projectFile, string projectFileOfTaskNode, int line, int column, string taskAssemblyLocation)
         {
             ErrorUtilities.VerifyThrow(targetBuildEventContext != null, "targetBuildEventContext is null");
-            BuildEventContext taskBuildEventContext = new BuildEventContext(
-                    targetBuildEventContext.SubmissionId,
-                    targetBuildEventContext.NodeId,
-                    targetBuildEventContext.ProjectInstanceId,
-                    targetBuildEventContext.ProjectContextId,
-                    targetBuildEventContext.TargetId,
-                    NextTaskId);
+            BuildEventContext taskBuildEventContext = targetBuildEventContext.WithTaskId(NextTaskId);
```

Comment:
Here's another example of losing an evalId on context creation - fixed with the new API